### DISCUSSION TITLE
Unable to start new AIO

### DISCUSSION THREAD

**Discussion by burnclouds** [nextcloud/all-in-one#2953] [Questions] [answered]
docker run --sig-proxy=false --name nextcloud-aio-mastercontainer --restart unless-stopped --publish 8080:8080 --publish 8443:8443 --publish 80:80 --volume /media/nc/nextcloud_aio_mastercontainer:/mnt/docker-aio-config --volume /var/run/docker.sock:/var/run/docker.sock:ro nextcloud/all-in-one:latest

use all defaults and enter my domain name... then:

![image](https://github.com/nextcloud/all-in-one/assets/9850267/e54db383-1b9a-4e6c-84b4-75bbf0a702d4)

"https://nc.domain.tld:8443/api/docker/logs?id=nextcloud-aio-nextcloud"
```
psql: error: connection to server at "nextcloud-aio-database" (172.18.0.4), port 5432 failed: FATAL:  password authentication failed for user "oc_nextcloud"
Waiting for the database to start...
psql: error: connection to server at "nextcloud-aio-database" (172.18.0.4), port 5432 failed: FATAL:  password authentication failed for user "oc_nextcloud"
Waiting for the database to start...

```

**Comment by burnclouds** [comment]
$ sudo docker exec nextcloud-aio-nextcloud grep dbpassword config/config.php
$ 

I presume the lack of a response means that the config file doesn't define the password properly (or at all) but I'm not a dbadmin. suggestions and Ideas are welcome.

**Comment by szaimen** [comment]
Hi, can you post the `nextcloud-aio-database` logs  here?

**Comment by burnclouds** [comment]
...:8443/api/docker/logs?id=nextcloud-aio-database:


Setting max connections...

PostgreSQL Database directory appears to contain a database; Skipping initialization

2023-07-14 12:07:12.716 UTC [13] LOG:  starting PostgreSQL 15.3 on x86_64-pc-linux-musl, compiled by gcc (Alpine 12.2.1_git20220924-r10) 12.2.1 20220924, 64-bit
2023-07-14 12:07:12.716 UTC [13] LOG:  listening on IPv4 address "0.0.0.0", port 5432
2023-07-14 12:07:12.716 UTC [13] LOG:  listening on IPv6 address "::", port 5432
2023-07-14 12:07:12.718 UTC [13] LOG:  listening on Unix socket "/var/run/postgresql/.s.PGSQL.5432"
2023-07-14 12:07:12.722 UTC [23] LOG:  database system was shut down at 2023-07-13 20:13:50 UTC
2023-07-14 12:07:12.726 UTC [13] LOG:  database system is ready to accept connections
2023-07-14 12:07:20.466 UTC [28] FATAL:  password authentication failed for user "oc_nextcloud"
2023-07-14 12:07:20.466 UTC [28] DETAIL:  Connection matched pg_hba.conf line 100: "host all all all scram-sha-256"
2023-07-14 12:07:25.482 UTC [30] FATAL:  password authentication failed for user "oc_nextcloud"
2023-07-14 12:07:25.482 UTC [30] DETAIL:  Connection matched pg_hba.conf line 100: "host all all all scram-sha-256"

**Comment by szaimen** [comment]
This looks like a hickup. Can you follow https://github.com/nextcloud/all-in-one#how-to-properly-reset-the-instance and try again?

**Comment by burnclouds** [comment, marked as answer]
Managed to get it working (I think). Turns out I'm an idiot, I changed "--volume nextcloud_aio_mastercontainer:/mnt/docker-aio-config" to "--volume /media/BTRFS/nextcloud:/mnt/docker-aio-config" thinking i was changing the storage location (I'm coming from nextcloudpi docker containers). So cleaned up everything per the above and actually RTFMd.... https://github.com/nextcloud/all-in-one#how-to-change-the-default-location-of-nextclouds-datadir

**Comment by karachungen** [comment]
to the people of future, i've managed to fix that issue by

- check **configuration.json** in the **nextcloud-aio-mastercontainer**. Look at **DATABASE_PASSWORD**
- check **config.php** file in the *nextcloud-aio-mastercontainer**. Check **dbpassword**
- Ensure that they are same
- If not replace password from **config.php** to **configuration.json** 
- Try to get authentificated using this credentials using **psql -h nextcloud-aio-database -U oc_nextcloud -d nextcloud_database**
- If it fails - login to **nextcloud-aio-database** as a **root**
- run **psql  -U oc_nextcloud -d nextcloud_database**
- ALTER password for user **oc_nextcloud** using password from **config.php** file

**Reply by rorar** [reply]
thanks to @karachungen 
To add up for everyone stumble upon this: 
If you want to keep your DB
- Backup your config.php and configuration.json to a safe place
- Delete the container and images
- don't remove the DB volume(!)
- remove every other volume than the DB volume
- remove the failed installation file in the root of your appdata folder


For reference (NC AiO @ Docker):

$ /var/lib/docker/volumes/nextcloud_aio_nextcloud/_data/config# nano config.php
$ /var/lib/docker/volumes/nextcloud_aio_mastercontainer/_data/data# nano configuration.json 

$ sudo docker volume ls --filter "dangling=true"
$ docker volume rm nextcloud_aio_apache nextcloud_aio_backup_cache nextcloud_aio_clamav nextcloud_aio_elasticsearch nextcloud_aio_mastercontainer nextcloud_aio_talk_recording nextcloud_aio_redis nextcloud_aio_nextcloud

$ docker exec -it --user root nextcloud-aio-database bash
$ ALTER USER oc_nextcloud WITH PASSWORD 'REPLACE_WITH_PASSWORT_FROM_CONFIG.PHP';
